src/modules/articles/components/extensions/MediaExtension.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+
+import { IExtensionProps } from './extensions';
+import { ArticleMedia } from '../ArticleMedia';
+
+interface IParsedProps {
+    mediumIds: string[]
+}
+
+export const MediaExtension: React.FunctionComponent<IExtensionProps> = ({ props, article }) => {
+    const parsedProps = props as IParsedProps;
+
+    //makes a copy so media filtering doesn't affect other components
+    article = Object.assign({}, article);
+    if (parsedProps.mediumIds && article.media) {
+        article.media = article.media.filter(m => parsedProps.mediumIds.indexOf(m.id) >= 0);
```

Comment:
Filter creates a new array so I don't think we need to copy it actually.

Reply:
Also, I'd convert `mediumIds` to a Set, just cause you're checking inclusion which is pretty inefficient in an array. Not a huge deal though.

Reply:
Since the most mediumIds there could possibly be is 15 in an extreme scenario, I think an array is fine. The copy is needed since if article.media were changed without the copy, the article all other components receive would be modified, meaning they wouldn't be able to render other media. (I had that exact bug which is why that line of code was introduced.)

Reply:
Ah okay. Object.assign does a shallow copy though, so idk how it manages to copy `article.media`. But if it works, it works.

Reply:
Implemented use of set anyway, avoiding quadratic complexity is a good goal although hopefully the cost of constructing a Set here isn't too great.